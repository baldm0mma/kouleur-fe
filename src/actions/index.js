export const setProjects = projects => ({
  type: 'SET_PROJECTS',
  projects
});

export const setPalettes = palettes => ({
  type: 'SET_PALETTES',
  palettes
});

export const setCurrentProject = (id, name) => ({
  type: 'SET_CURRENT_PROJECT',
  id,
  name
});

export const setError = error => ({
  type: 'SET_ERROR',
  error
});

export const toggleNewProject = boolean => ({
  type: 'TOGGLE_NEW_PROJECT',
  boolean
})
