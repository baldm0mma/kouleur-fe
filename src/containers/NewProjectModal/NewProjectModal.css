.modal-wrapper{
  position: absolute;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;

  }
.modal-card{
  position: fixed;
  background-color: white;
  border-radius: 5px;
  padding: 15px;
  z-index: 10;
  box-shadow: 2px 2px 10px rgba(0,0,0, 0.3);
  width: 750px;
  display: block;
}

.background {
  position: fixed;
  width: 100vw;
  height: 100vw;
  top: 0;
  left: 0;
  background: black;
  opacity: 0.6;
}