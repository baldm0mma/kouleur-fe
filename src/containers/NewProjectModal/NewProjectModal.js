import React, { Component } from 'react';
import './NewProjectModal.css';
import { connect } from 'react-redux';
import { setCurrentProject, toggleNewProject, setError, setProjects } from '../../actions';
import { getAllProjectsUrl } from '../../utilities/urls';
import { Button, Form } from 'semantic-ui-react';
import { postProjectName, getProjects } from '../../utilities/apiCalls';

export class NewProjectModal extends Component {
  constructor() {
    super();
    this.state = {
      project_name: ''
    };
  }

  handleNameChange = e => {
    this.setState({ project_name: e.target.value });
  };

  makeNewProject = async () => {
    const name = this.state.project_name;
    try {
      const response = await postProjectName(getAllProjectsUrl, name);
      this.props.setCurrentProject(response, name);
      const projects = await getProjects(getAllProjectsUrl);
      await this.props.setProjects(projects);
      this.props.toggleNewProject(false);
    } catch (error) {
      this.props.setError(error.message);
    }
  };

  render() {
    return (
      <article className='modal-wrapper'>
        <div className='modal-card'>
          <Form>
            <Form.Field>
              <label id='modal-title'>Project Name</label>
              {this.props.errorMessage && <p>Sorry, there was an issue posting: {this.props.errorMessage}</p>}
              <input
                placeholder='Project Name'
                value={this.state.project_name}
                onChange={e => this.handleNameChange(e)}
              />
            </Form.Field>
            <Button
              id='create-project-button'
              type='submit'
              onClick={this.makeNewProject}
            >
              Create Project
            </Button>
            <Button
              onClick={() => this.props.toggleNewProject(false)}
              type='submit'
            >
              Cancel
            </Button>
          </Form>
        </div>
        <div className='background' />
      </article>
    );
  }
}

export const mapStateToProps = ({ projects, errorMessage }) => ({
  projects,
  errorMessage
});

export const mapDispatchToProps = dispatch => ({
  setCurrentProject: (id, name) => dispatch(setCurrentProject(id, name)),
  toggleNewProject: boolean => dispatch(toggleNewProject(boolean)),
  setError: error => dispatch(setError(error)),
  setProjects: results => dispatch(setProjects(results))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(NewProjectModal);
