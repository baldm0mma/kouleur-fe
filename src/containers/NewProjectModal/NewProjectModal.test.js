import React from 'react';
import { shallow } from 'enzyme';
import {mapStateToProps, mapDispatchToProps, NewProjectModal} from './NewProjectModal.js'
import { setCurrentProject, toggleNewProject, setError, setProjects } from '../../actions';


describe('New Project Modal',() => {
  let wrapper;
  let projects;
  let mockDispatch
  beforeEach(() => {
    projects = [{project_name: 'evans fun proj', id: 2}]
    wrapper = shallow(<NewProjectModal projects={projects} errorMessage={''}/>)
    mockDispatch = jest.fn()
    setCurrentProject=jest.fn()
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should map state to props', () => {
    const mappedProps = mapStateToProps({projects: projects}, {errorMessage: ''});
    expect(mappedProps).toEqual({projects: projects}, {errorMessage: ''});
  })
  it('should map setCurrentProject to props', () => {
    const mockAction = setCurrentProject(projects[0]);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.setCurrentProject(projects[0])
    expect(mockDispatch).toHaveBeenCalledWith(mockAction);
  })
   it('should map toggleNewProject to props', () => {
    const mockAction = toggleNewProject(false);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.toggleNewProject(false)
    expect(mockDispatch).toHaveBeenCalledWith(mockAction);
  })
  it('should map setError to props', () => {
    const mockAction = setError('error');
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.setError('error')
    expect(mockDispatch).toHaveBeenCalledWith(mockAction);
  })
  it('should map setError to props', () => {
    const mockAction = setProjects(projects);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.setProjects(projects);
    expect(mockDispatch).toHaveBeenCalledWith(mockAction);
  })
  it.skip('should call setCurrentProject after makeNewProject', () => {
    wrapper.instance().makeNewProject()
    expect(setCurrentProject).toHaveBeenCalled();
  })

})