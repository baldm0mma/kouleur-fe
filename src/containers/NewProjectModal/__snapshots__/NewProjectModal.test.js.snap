// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Project Modal should match the snapshot 1`] = `ShallowWrapper {}`;
